consensus/doc/PROTOCOL_STATE_SYNC.md

```diff
@@ -0,0 +1,119 @@
+
+In `SyncedMode`, the Node is caught up to the latest block and relies on new blocks to be propagated via the P2P network after Validators reach consensus.
+
+### Sync Mode
+If the Node is `Syncing` or `localSyncState.Height < GlobalSyncMeta.Height` then the `StateSync` protocol is in `SyncMode`.
+
+In `SyncMode`, the Node is catching up to the latest block by making `BlockRequests` to its fellow eligible peers. A peer is eligible for a `BlockRequest` if `PeerMeta.MinHeight` <= `self.MaxBlockHeight` <= `PeerMeta.MaxHeight`.
+
+Though it is `unspecified` whether or not a Node may make `BlockRequests` in order or in parallel, the cryptographic restraints of block processing require the Node to call `ApplyingBlock` sequentially until it is `Synced`.
+
+It is important to note, if any blocks processed result in an invalid `AppHash` during `ApplyBlock`, a new `BlockRequest` must be issued until a valid block is found.
+
+### Server Mode
+
+In the `StateSync` protocol, the Node fields valid `BlockRequests` from its peers to help them `CatchUp` to be `Synced`. This sub-protocol is continuous throughout the lifecycle of StateSync.
+
+```mermaid
+graph TD
+    A[StateSync] -->|IsCaughtUp| B(Synced Mode)
+    B --> |Consensus Messages| C(ConsensusModule.Synced)
+    A --> |IsSyncing| E(Sync Mode)
+    E --> |Request Block| G[Peers]
+    G--> |ApplyBlock| A
+    A --> D[Server Mode]
+    D --> |Serve Blocks Upon Request| G
+```
+
+## Follow up tasks
+
+* `Fast Sync Design` - Sync only the last `N` blocks from `Latest Network Height`
+
+* `Optimistic Sync Design` - Optimize the State Sync protocol by parallelling requests and computation
+
+* `Block Chunk Design` - Update the Block by Block design to be able to receive and provide multiple blocks at a time.
+
+* `Block Stream Design` - Update the Block by Block design to stream blocks via a WebSocket from a single connectable peer.
+
+## Research Items
+TODO(M5): Create issues to track and discuss these work items.
+
+How the persistence layer design of `pruning` the Merkle Tree affects `StateSync`.
+
+How DB Pruning of the `SQL DB` might affect fast sync.
+
+How the Churn Management operations of RainTree might provide opportunities and obstacles with StateSync.
+
+## Glossary
+
+`ApplyingBlock`: The process of playing block parts and its subsequent transactions against the Node's world state using the Utility Module and Validating the `AppHash` contained in the block against the produced `AppHash` from the local state.
+
+`BlockRequests`: A message from an active peer, requesting a block to sync the chain to the Global Network State.
+
+`Churn Management Protocol`: The protocol in Pocket's P2P Module that ensures the most updated and valid Network Peer list possible.
+
+`Network Peer`: A node that is directly connected and sharing resources without going through a third party server. Peers may start the connection through an `inbound` or `outbound` initialization.
```

Comment:
I feel like `resources` can be interpreted as compute resources so I rewarded it a bit.

```suggestion
`Network Peer`: Another node on the network that this node can directly communicate with, without going through a third-party server. Peers may start the connection through an `inbound` or `outbound` initialization to share and transmit data.
```